bids/layout/writing.py

```diff
@@ -214,8 +214,11 @@ def write_contents_to_file(path, contents=None, link_to=None,
         to the new path.
     link_to : str
         Optional path with which to create a symbolic link to.
-        Used as an alternative to and takes priority over the contents
+        Used as an alternative to, and takes priority over, the contents
         argument.
+    copy_from : str
+        Optional filename to copy to new location. Used an alternative to, and
+        takes priority over, the contents argument.
```

Comment:
why to occlude "write_to_file" with "link_file" and "copy_file" and introducing all kinds of interaction now complicating description of how this thing works with all the "priority over"?  Note also that `link_to` takes "priority over" "copy_from".  Why not to keep this function without any `link_to`, and `copy_from` and have dedicated helpers (khe khe -- `shutils.copy`)  for specific linking/copying operations?
If that was done to outside avoid needing to duplicate the string for the path or introducing a variable, may be this and those `link_file`, `copy_file` could just return the filename, so invocation could then be chained like
`copy_file(
    write_file("mylongfilename", contents=...),  # would return mylongfilename
    "newlongfilename"
)`
if really desired.   Would also allow to copy and link at the same time ;) 
but now I see that all that takes priority over contents argument -- so why it is all in one function?

Reply:
BTW, having helpers even if only lean shims around `shutils.copy` might come handy... At some point some smart folks should implement support for CoW filesystems (like BTRFS which I use) for instant copies of huge files.  so having a single point to change would be of help.

Reply:
The answer is just expediency. I don't have the time to rewrite all of this stuff more cleanly, but I did have 5 minutes to spend patching it to work more generally. And note that we still benefit from all the conflict resolution stuff inside the original method.

If you want to refactor this so it makes more sense, a PR would be great!